### PR TITLE
LEGAL-35 website: Make CORS allow static files form srk origins

### PR DESCRIPTION
**Pull request by kirelagin**
We want to reuse some of the website’s static data (e.g. fonts) on other
pages. By default, CORS is enforced for fonts, so we need to explicitly
allow the origins that we want.

Make nginx send Allow Origin if the Origin is *.serokell.io.

Unfortunately, this header does not allow wildcards, so the standard
trick is to match on the value of the Origin header and, if it is good,
set Allow Origin to the value of Origin.

Doubly unfortunately, `if` in nginx `location` blocks does not
really work, so, insead, we match on the origin and set the $srk_origin
variable on the top level (in the `http` block), and then just set Allow
Origin to the value of this variable (which will be empty if Origin did
not match).

### CODE REVIEW

**Comment by kirelagin** [comment]
Someone, pls, review this and, if everything is ok, merge and deploy ASAP. Thanks!

File: lib/modules/website.nix [hunk 1]
```diff
@@ -85,7 +94,13 @@ in {
       '';
 
       locations = {
-        "/static".extraConfig = "expires 7d;";
+        "/static".extraConfig = ''
+          expires 7d;
+
+          # CORS: Allow from Serokell origins. The spec does not allow wildcards :/.
+          add_header 'Access-Control-Allow-Origin' $srk_origin;
```
Comment (notgne2) [inline lib/modules/website.nix:101]: is this the only CORS header that we need? not anything for methods, etc?

**Review by notgne2 — COMMENTED** [review]
First thought that popped into my head is "this header will get cached and break everything" but iirc that's what the "Vary" header should prevent that in theory, but are we absolutely sure it will? I'd feel more comfortable if there's another source for that working besides general intuition, I did a quick search and didn't see anybody suggest it though

File: lib/modules/website.nix (hunk 1 above)
Comment (cab404) [reply lib/modules/website.nix:101]: Nope, no methods required.

Comment (cab404) [reply lib/modules/website.nix:101]: The server must not specify the "*" wildcard for the Access-Control-Allow-Methods response-header value, but must instead specify an explicit list of method names; for example, Access-Control-Allow-Methods: POST, GET

Comment (cab404) [reply lib/modules/website.nix:101]: as per https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Review by cab404 — CHANGES_REQUESTED** [review]

**Comment by kirelagin** [comment]
@cab404 First, `Allow-Methods` is only used in preflighted requests, the request I am interested in is not preflighted (I don’t think simple GET requests – the only kind we are really interested in for static content – ever get preflighted).

Second, the part that you cites is about preflighted _credential_ requests. I have absolutely no idea what it is, but definitely not something that we are interested in for this use-case. (Lastly, I don’t specify `*`, I simply do not specify anything, so what you cited does not apply.)

**Comment by kirelagin** [comment]
(The real reasons I don’t want to do this is that (a) I don’t know what methods to list... just GET? (b) This require another map with copy-pasted pattern and so on.)

**Review by notgne2 — APPROVED** [review]
I think this is fine, on the basis of the part of this https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#the_http_response_headers about "Vary"

**Comment by kirelagin** [comment]
Ok, for posterity (regarding `Vary`). This is the spec [explaining why it is needed and enough](https://fetch.spec.whatwg.org/#cors-protocol-and-http-caches). Here is a [very popular SO answer](https://stackoverflow.com/a/27990162/603094).

**Pull request merged**